### PR TITLE
[MIG] l10n_ar_account_reports: checks_to_date

### PR DESCRIPTION
**Pull request by ica-adhoc**

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina-ee/pull/188).

File: l10n_ar_account_reports/__manifest__.py [hunk 1]
```diff
@@ -31,7 +31,12 @@
     'depends': [
         'account_reports',
```
Comment (zaoral) [inline l10n_ar_account_reports/__manifest__.py:32]: also need to update the manifest version

File: l10n_ar_account_reports/reports/report_checks_to_date.xml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            </t>
+            <div class="page">
+                <br/>
+                <h5>Cheques propios</h5>
+                <table class="table table-sm o_main_table table-bordered">
+                    <thead>
+                        <tr>
+                            <th class="text-left">Número</th>
+                            <th class="text-left">Fecha contable</th>
+                            <!-- <th class="text-left">Fecha emisión</th> -->
```
Comment (zaoral) [inline l10n_ar_account_reports/reports/report_checks_to_date.xml]: remove this comment, that field does not exist anymore

File: l10n_ar_account_reports/reports/report_checks_to_date.xml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                            <th class="text-left">Fecha de pago</th>
+                            <th class="text-left">Empresa</th>
+                            <th class="text-left">Chequera</th>
```
Comment (zaoral) [inline l10n_ar_account_reports/reports/report_checks_to_date.xml:22]: take into account when pssing this to 16that chequera does not exist anymore

File: l10n_ar_account_reports/reports/report_checks_to_date.xml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        </t>
+    </t>
+</template>
+
+<record id="checks_to_date_report" model="ir.actions.report">
+    <field name="name">Cheques a fecha</field>
+    <field name="model">account.check.to_date.report.wizard</field>
+    <field name="report_type">qweb-pdf</field>
+    <field name="report_name">l10n_ar_account_reports.checks_to_date</field>
+    <field name="report_file">l10n_ar_account_reports.checks_to_date</field>
```
Comment (zaoral) [inline l10n_ar_account_reports/reports/report_checks_to_date.xml:115]: this is the name of the generated pdf? if it is. maybe a Checks To Date xxxx-xx-xx could be a god idea as file name

File: l10n_ar_account_reports/wizards/checks_to_date_report.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        'Hasta Fecha',
+        required=True,
+        default=fields.Date.today,
+    )
+
+    def action_confirm(self):
+        self.ensure_one()
+        force_domain = self.journal_id and [('journal_id', '=', self.journal_id.id)] or []
+
+        # cheques de tercero en mano
```
Comment (zaoral) [inline l10n_ar_account_reports/wizards/checks_to_date_report.py]: remove commented code

File: l10n_ar_account_reports/wizards/checks_to_date_report.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        query = """
+                SELECT DISTINCT ON (t.check_id) t.check_id AS cheque FROM
+                    (
+                        SELECT ap.id as check_id, ap_move.date as operation_date, apm.code as operation_code
+                        FROM account_payment ap
+                        LEFT JOIN account_payment_method AS apm ON apm.id = ap.payment_method_id
+                        LEFT JOIN account_move AS ap_move ON ap.move_id = ap_move.id
+                        LEFT JOIN account_journal AS journal ON ap_move.journal_id = journal.id
+                        WHERE
+                        apm.code = 'check_printing' AND ap_move.date <= '""" + to_date + """' order by ap.id, ap_move.date desc, ap.id desc
```
Comment (zaoral) [inline l10n_ar_account_reports/wizards/checks_to_date_report.py]: you could also use 
```suggestion
                        apm.code = 'check_printing' AND ap_move.date <= $s order by ap.id, ap_move.date desc, ap.id desc
```

and then at the end of the SQL string add """ % (to_date)

File: l10n_ar_account_reports/wizards/checks_to_date_report.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+                        SELECT ap.id as check_id, afr_full.name as conciliation_name, aml_2.date as operation_date, aml.id as aml_1, aml_2.id as aml_2
+                        FROM account_payment ap
+                        JOIN account_payment_method AS apm ON apm.id = ap.payment_method_id
+                        JOIN account_move_line as aml ON ap.move_id = aml.move_id
+                        JOIN account_full_reconcile AS afr_full ON afr_full.id = aml.full_reconcile_id
+                        JOIN account_move_line AS aml_2 ON aml_2.full_reconcile_id = afr_full.id
+                        WHERE apm.code = 'check_printing' AND aml.id <> aml_2.id
+                    ) t2
+                    ON t.check_id = t2.check_id
+                WHERE t2.operation_date >= '""" + to_date + """' OR t2.operation_date IS NULL
```
Comment (zaoral) [inline l10n_ar_account_reports/wizards/checks_to_date_report.py]: same of above

**Review by zaoral — COMMENTED** [review]
LGTM
Dont forget

* change WIP name of th pr.
* squash all the commits
* rebase branch because is out of date
* move the commented code

**Comment by zaoral** [comment]
@roboadhoc r+

**Comment by fw-bot-adhoc** [comment]
@ica-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#198

**Comment by fw-bot-adhoc** [comment]
@ica-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#198

**Comment by fw-bot-adhoc** [comment]
@ica-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#198

**Comment by fw-bot-adhoc** [comment]
@ica-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#198

**Comment by fw-bot-adhoc** [comment]
@ica-adhoc @zaoral this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#198

**Pull request closed without merging**